### PR TITLE
ACM-8973 Update API description

### PR DESCRIPTION
**Pull request by fxiang1**
https://issues.redhat.com/browse/ACM-8973

- Update API description

### CODE REVIEW

File: deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml [hunk 1]
```diff
@@ -308,21 +312,21 @@ spec:
...
                           description: A comma-separated list of tags to specify which
-                            sets of ansible tasks in a job should not be run.
+                            sets of Ansible tasks in a job should not be run.
                           type: string
                         type:
                           default: Job
                           description: Type of the Hook. For Job type, Ansible job
                             template will be used. For Workflow type, Ansible workflow
-                            template will be used. If omitted, default to Job type.
+                            template will be used. If omitted, defaults to Job type.
```
Comment (jc-berger) [inline deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml:329]:
```suggestion
                            template is not used. If omitted, it defaults to the Job type.
```

File: deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml [hunk 2]
```diff
@@ -332,18 +336,18 @@ spec:
...
                   towerAuthSecret:
-                    description: TowerAuthSecret is ansible secret used in template
-                      to run in tower
+                    description: TowerAuthSecret is an Ansible secret used in the
+                      template to provide authentication to an Ansbile tower.
                     type: string
                 type: object
               upgrade:
-                description: During an upgrade curation run these
+                description: During an upgrade curation run these hooks.
```
Comment (jc-berger) [inline deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml:344]:
```suggestion
                description: An upgrade curation runs these hooks.
```

File: deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml [hunk 3]
```diff
@@ -332,18 +336,18 @@ spec:
...
                 properties:
                   channel:
                     description: Channel is an identifier for explicitly requesting
                       that a non-default set of updates be applied to this cluster.
-                      The default channel will be contain stable updates that are
-                      appropriate for production clusters.
+                      The default channel will contain stable updates that are appropriate
```
Comment (jc-berger) [inline deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml:349]:
```suggestion
                      The default channel contains stable updates that are appropriate
```

File: deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml [hunk 4]
```diff
@@ -362,17 +366,17 @@ spec:
                   monitorTimeout:
                     default: 120
                     description: MonitorTimeout defines the monitor process timeout,
-                      the unit of this is minute. By default, it is 120 minutes If
+                      the unit of this is minute. By default, it is 120 minutes. If
```
Comment (jc-berger) [inline deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml:369]:
```suggestion
                      and defines time in minutes. By default, it is 120 minutes. If
```

File: deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml [hunk 5]
```diff
@@ -382,21 +386,21 @@ spec:
...
                           description: A comma-separated list of tags to specify which
-                            sets of ansible tasks in a job should not be run.
+                            sets of Ansible tasks in a job should not be run.
                           type: string
                         type:
                           default: Job
                           description: Type of the Hook. For Job type, Ansible job
                             template will be used. For Workflow type, Ansible workflow
-                            template will be used. If omitted, default to Job type.
+                            template will be used. If omitted, defaults to Job type.
```
Comment (jc-berger) [inline deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml:403]:
```suggestion
                            template is used. If omitted, it defaults to the Job type.
```

File: deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml [hunk 6]
```diff
@@ -416,21 +420,21 @@ spec:
...
```
Comment (jc-berger) [inline deploy/crd/cluster.open-cluster-management.io_clustercurators.yaml:437]:
```suggestion
                            template is used. If omitted, it defaults to the Job type.
```

File: pkg/api/v1beta1/clustercurator_types.go [hunk 7]
```diff
@@ -14,44 +14,46 @@ type ClusterCuratorSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// This is the desired curation that will occur
+	// This is the desired curation that will occur. The supported options are 'install', 'upgrade', or 'destroy'.
```
Comment (jc-berger) [inline pkg/api/v1beta1/clustercurator_types.go:17]:
```suggestion
	// This is the desired curation that occurs. The supported options are 'install', 'upgrade', or 'destroy'.
```

File: pkg/api/v1beta1/clustercurator_types.go [hunk 8]
```diff
@@ -14,44 +14,46 @@ type ClusterCuratorSpec struct {
...
 	// +kubebuilder:validation:Enum={install,scale,upgrade,destroy,delete-cluster-namespace}
 	DesiredCuration string `json:"desiredCuration,omitempty"`
 
 	// Points to the Cloud Provider or Ansible Provider secret, format: namespace/secretName
 	ProviderCredentialPath string `json:"providerCredentialPath,omitempty"`
 
-	// During an install curation run these Pre/Post hooks
+	// During an install curation run these pre/post hooks.
```
Comment (jc-berger) [inline pkg/api/v1beta1/clustercurator_types.go:24]:
```suggestion
	// During an install curation run these prehooks and posthooks.
```

File: pkg/api/v1beta1/clustercurator_types.go [hunk 9]
```diff
@@ -14,44 +14,46 @@ type ClusterCuratorSpec struct {
...
 	Install Hooks `json:"install,omitempty"`
 
-	// During an scale curation run these Pre/Post hooks
+	// During an scale curation run these pre/post hooks.
```
Comment (jc-berger) [inline pkg/api/v1beta1/clustercurator_types.go:27]:
```suggestion
	// During an scale curation, run these prehooks and posthooks.
```

File: pkg/api/v1beta1/clustercurator_types.go [hunk 10]
```diff
@@ -14,44 +14,46 @@ type ClusterCuratorSpec struct {
...
 	Scale Hooks `json:"scale,omitempty"`
 
-	// During an destroy curation run these **Pre hook ONLY**
+	// During an destroy curation run these hooks.
```
Comment (jc-berger) [inline pkg/api/v1beta1/clustercurator_types.go:30]:
```suggestion
	// During an destroy curation, run these hooks.
```

File: pkg/api/v1beta1/clustercurator_types.go [hunk 11]
```diff
@@ -14,44 +14,46 @@ type ClusterCuratorSpec struct {
...
+	// Standalone clusters only support the prehook.
+	// Hosted clusters support both prehook/posthook.
```
Comment (jc-berger) [inline pkg/api/v1beta1/clustercurator_types.go:32]:
```suggestion
	// Hosted clusters support both prehook and posthook.
```

File: pkg/api/v1beta1/clustercurator_types.go [hunk 12]
```diff
@@ -14,44 +14,46 @@ type ClusterCuratorSpec struct {
...
 type Hook struct {
-	// Name of the Ansible Template to run in Tower as a job
+	// Name of the Ansible Template to run in the Ansible Tower as a job.
 	// +kubebuilder:validation:Required
 	Name string `json:"name"`
 
 	// Type of the Hook. For Job type, Ansible job template will be used.
 	// For Workflow type, Ansible workflow template will be used.
-	// If omitted, default to Job type.
+	// If omitted, defaults to Job type.
```
Comment (jc-berger) [inline pkg/api/v1beta1/clustercurator_types.go:56]:
```suggestion
	// If omitted, it defaults to the Job type.
```

**Comment by fxiang1** [comment]
@jc-berger I made the changes. Please take a look again when you get a chance. Thanks!

**Review by jc-berger — APPROVED** [review]
/lgtm

**Pull request merged**